src/Dialog.cc

```diff
@@ -101,37 +103,43 @@ bool Dialog::UpdateConfigAttribute(const std::string &_path,
   }
 
   // Update attribute value for the correct dialog
+  bool updated{false};
   for (auto dialogElem = doc.FirstChildElement("dialog");
     dialogElem != nullptr;
     dialogElem = dialogElem->NextSiblingElement("dialog"))
   {
-    if(dialogElem->Attribute("name") == this->objectName().toStdString())
+    if (dialogElem->Attribute("name") == this->objectName().toStdString())
     {
       dialogElem->SetAttribute(_attribute.c_str(), _value);
+      updated = true;
     }
   }
 
-  // Write config file
-  tinyxml2::XMLPrinter printer;
-  doc.Print(&printer);
+  // Create new <dialog> if missing
+  if (!updated)
+  {
+    auto dialogElem = doc.NewElement("dialog");
+    dialogElem->SetAttribute("name", this->objectName().toStdString().c_str());
+    dialogElem->SetAttribute(_attribute.c_str(), _value);
+    doc.InsertEndChild(dialogElem);
+  }
 
-  std::string config = printer.CStr();
-  std::ofstream out(_path.c_str(), std::ios::out);
-  if (!out)
+  // Write config file
+  if (!doc.SaveFile(_path.c_str()))
   {
-    ignerr << "Unable to open file: " << _path
+    // LCOV_EXCL_START
+    ignerr << "Failed to save file: " << _path
            << ".\nCheck file permissions.\n";
+    // LCOV_EXCL_STOP
   }
-  else
-    out << config;
 
   return true;
 }
 
 /////////////////////////////////////////////////
-void Dialog::SetDefaultConfig(const std::string &_config)
+void Dialog::SetDefaultConfig(const std::string &)
 {
-  this->dataPtr->config = _config;
+  ignwarn << "Dialog::SetDefaultConfig has no effect." << std::endl;
```

Comment:
This means the only way to write to the dialog config is to use the `Update` function, right?

Reply:
I imagine if the config has many fields it might be not the best way, or is that not a major use case?

Reply:
Yeah I believe that's the only way right now. If the user wants to set multiple attributes, they'd need to set them one by one. And there's no interface for adding elements, etc.